packages/block-renderer/src/hooks/use-pattern-inline-css.tsx

```diff
@@ -0,0 +1,53 @@
+import { shuffle } from '@automattic/js-utils';
+import { useMemo } from 'react';
+
+// Shuffle order of blog posts to avoid repetition in previews
+const shufflePostOrder = ( blogPostCount: number ) =>
+	shuffle( [ ...Array( blogPostCount ).keys() ] );
+
+// Memoize order per pattern to show same order in previews
+const inlineCssByPatternId: { [ key: string ]: string } = {};
+
+// Memoize last order
+let lastPostOrder: number[] = [];
+
+const usePatternInlineCss = ( patternId: string, patternHtml: string, isNewSite: boolean ) => {
+	return useMemo( () => {
+		const hasGrid = patternHtml?.includes( 'is-layout-grid' );
+		const blogPostCount = patternHtml?.match( /wp-block-post /g )?.length ?? 0;
+
+		// Only for patterns with a grid of posts in newly created sites
+		if ( ! isNewSite || ! hasGrid || blogPostCount === 1 ) {
```

Comment:
I think this still has a problem. We also need to exit early if the site has no posts at all...

Reply:
Okay, let's add that check and return early :)

I wanted to note that `blogPostCount` does not count the site posts but the pattern posts and only for newly created sites, which means the post from Creatio demo site.

In short, `blogPostCount` won't be zero because Creatio demo site will always have posts.